source/Halibut.Tests/Transport/Streams/AsyncStreamFixture.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support.Streams;
+using Halibut.Tests.Util;
+using Halibut.Transport.Streams;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Transport.Streams
+{
+    public class AsyncStreamFixture : StreamWrapperSupportsAsyncIOFixture
+    {
+        protected override AsyncStream WrapStream(Stream stream) => new NoOpAsyncStream(stream);
+    }
+
+    public class NoOpAsyncStream : AsyncStream
+    {
+        readonly Stream stream;
+
+        public NoOpAsyncStream(Stream stream)
+        {
+            this.stream = stream;
+        }
+
+        public override void Flush()
+        {
+            stream.Flush();
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return stream.Read(buffer, offset, count);
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            return stream.Seek(offset, origin);
+        }
+
+        public override void SetLength(long value)
+        {
+            stream.SetLength(value);
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            stream.Write(buffer, offset, count);
+        }
+
+        public override bool CanRead => stream.CanRead;
+
+        public override bool CanSeek => stream.CanSeek;
+
+        public override bool CanWrite => stream.CanWrite;
+
+        public override long Length => stream.Length;
+
+        public override long Position
+        {
+            get => stream.Position;
+            set => stream.Position = value;
+        }
+
+        protected override Task<int> _ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return stream.ReadAsync(buffer, offset, count, cancellationToken);
+        }
+
+        protected override Task _WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return stream.WriteAsync(buffer, offset, count, cancellationToken);
+        }
+
+        protected override Task _FlushAsync(CancellationToken cancellationToken)
+        {
+            return stream.FlushAsync(cancellationToken);
+        }
+
+        protected override ValueTask _DisposeAsync()
+        {
+            return stream.DisposeAsync();
+        }
+    }
+
+    public abstract class StreamWrapperSupportsAsyncIOFixture : BaseTest
```

Comment:
Nit: I think this would be cleaner in a new file

Reply:
Done here https://github.com/OctopusDeploy/Halibut/pull/457